src/app/jscaip/HexaDirection.ts

```diff
@@ -58,11 +55,13 @@ export class HexaDirection extends Direction {
         }
     }
 
-    private constructor(x: 0|1|-1, y: 0|1|-1) {
+    private constructor(x: -2|-1|0|1|2, y: -2|-1|0|1|2) {
         super(x, y);
     }
-    public getOpposite(): HexaDirection {
+
+    public getOpposite(): this {
         const opposite: MGPFallible<HexaDirection> = HexaDirection.factory.from(-this.x, -this.y);
-        return opposite.get();
+        return opposite.get() as this;
```

Comment:
hmmm pourquoi, this == HexaDirection tout le temps ici non ?

Reply:
![image](https://github.com/EveryBoard/EveryBoard/assets/41200522/e82816ab-fa67-41d9-8882-8db5a627c60c)

Reply:
ok bah alors c'est pas safe, ça veut dire qu'on va se retrouver un HexaDirection casté en DodecaHexaDirection, c'qui est haram